### ISSUE TITLE
Unexpected error when using a decltype(auto) NTTP in an out-of-line class template member definition

### ISSUE THREAD

**Issue by miyuki** [llvm/llvm-project#56707]
Consider the following code:

```C++
template<decltype(auto) V>
struct S {
    int f();
};

template<decltype(auto) V>
int S<V>::f() { return 1; }
```

Clang, starting with v 9.0 and up to the current trunk version, rejects the code:
```console
$ clang -std=c++17 test.cc
<source>:7:11: error: nested name specifier 'S<V>::' for declaration does not refer into a class, class template or class template partial specialization
int S<V>::f() { return 1; }
    ~~~~~~^
```

Clang 8 and recent versions of GCC accept the code without errors.

**Comment by amilendra** [comment]
I believe this has been fixed by https://github.com/llvm/llvm-project/pull/89862. Closing
